src/WebSocketsServiceProvider.php

```diff
@@ -23,30 +26,47 @@
 
 class WebSocketsServiceProvider extends ServiceProvider
 {
-    public function boot(BroadcastManager $broadcastManager)
+    public function boot()
     {
         $this->publishes([
-            __DIR__.'/../config/websockets.php' => base_path('config/websockets.php'),
+            __DIR__ . '/../config/websockets.php' => base_path('config/websockets.php'),
         ], 'config');
 
-        if (! class_exists('CreateWebSocketsStatisticsEntries')) {
+        if (!class_exists('CreateWebSocketsStatisticsEntries')) {
             $this->publishes([
-                __DIR__.'/../database/migrations/create_websockets_statistics_entries_table.php.stub' => database_path('migrations/'.date('Y_m_d_His', time()).'_create_websockets_statistics_entries_table.php'),
+                __DIR__ . '/../database/migrations/create_websockets_statistics_entries_table.php.stub' => database_path('migrations/' . date('Y_m_d_His', time()) . '_create_websockets_statistics_entries_table.php'),
             ], 'migrations');
         }
 
         $this
             ->registerRoutes()
             ->registerDashboardGate();
 
-        $this->loadViewsFrom(__DIR__.'/../resources/views/', 'websockets');
+        $this->loadViewsFrom(__DIR__ . '/../resources/views/', 'websockets');
 
         $this->commands([
             Console\StartWebSocketServer::class,
             Console\CleanStatistics::class,
         ]);
 
-        $broadcastManager->extend('redis-pusher', function ($app, array $config) {
+        $this->configurePubSub();
+
+    }
+
+    protected function configurePubSub()
+    {
+        if (config('websockets.replication.enabled') !== true || config('websockets.replication.driver') !== 'redis') {
+            $this->app->singleton(ReplicationInterface::class, function () {
+                return (new EmptyClient());
+            });
+            return;
+        }
+
+        $this->app->singleton(ReplicationInterface::class, function () {
+            return (new RedisClient())->boot($this->loop);
+        });
+
+        app(BroadcastManager::class)->extend('redis-pusher', function ($app, array $config) {
```

Comment:
We have `$this->app`, can do `$this->app->get(BroadcastManager::class)` instead probably. (or make() instead of get()? not sure) but I generally prefer to use DI instead /shrug